consensus/tests/test_helper.go

```diff
@@ -442,8 +462,19 @@ func CreateBlock(blockchain *BlockChain, chainConfig *params.ChainConfig, starti
 			for _, v := range masternodesFromV1LastEpoch {
 				header.Validators = append(header.Validators, v[:]...)
 			}
+		} else if roundNumber%int64(chainConfig.XDPoS.Epoch) == 0 {
+			// epoch switch blocks, copy the master node list and inject into v2 validators
+			// Get last master node list from last v1 block
+			lastv1Block := blockchain.GetBlockByNumber(chainConfig.XDPoS.XDPoSV2Block.Uint64())
+			masternodesFromV1LastEpoch := decodeMasternodesFromHeaderExtra(lastv1Block.Header())
+			for _, v := range masternodesFromV1LastEpoch {
+				header.Validators = append(header.Validators, v[:]...)
+			}
+			// inject penalty for testing penalty hook if chainId is the specific one 1338
+			if chainConfig.ChainId.Int64() == 1338 {
+				header.Penalties = append(header.Penalties, signer[:]...)
+			}
```

Comment:
either we have penalties in all test or we can insert this penalties header in specific test.